indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -365,6 +365,7 @@ public TaskStatus call()
                                   )
                               );
 
+                              command.add(String.format("-Ddruid.indexer.taskId=%s", task.getId()));
```

Comment:
Interesting - I noticed some metrics related properties being set with the taskId but was hesitant to use them because I didn't want someone changing them not knowing they were being used for non-metric purposes as well. But being able to utilize `DataSourceTaskIdHolder` feels better to me, thanks!